lib/protobuf/field/field_hash.rb

```diff
@@ -0,0 +1,97 @@
+module Protobuf
+  module Field
+    class FieldHash < Hash
+
+      ##
+      # Attributes
+      #
+
+      attr_reader :field, :key_field, :value_field
+
+      ##
+      # Constructor
+      #
+
+      def initialize(field)
+        @field = field
+        @key_field = field.type_class.get_field(:key)
+        @value_field = field.type_class.get_field(:value)
+      end
+
+      ##
+      # Public Instance Methods
+      #
+
+      def []=(key, val)
+        super(normalize_key(key), normalize_val(val)) unless val.nil?
```

Comment:
I don't think that `nil` should ever be a valid value for any protobuf type so not allowing things to be nil is ok. Throwing an error might be better?

@nerdrew don't forget that you will still be able to use `delete` if you want to `delete` a field in your hash.